### ISSUE TITLE
Sketch generation - 1st code reviewing

### ISSUE THREAD

**Issue by quang-anh-nguyen** [dataflowr/Project-DL-Seq2Seq#1]
- As in the paper [page 2] each sequence of strokes is represented by an array of 5-elements array (dx, dy, p1, p2, p3), where (p1,p2,p3) is hotkey embedding: p1==1 if the stroke is drawn to the next point, p2==1 if not, p3==1 if drawing is finished. However, the function 'draw_image' from 'eval_skrnn.py' seems to inverse the roles of p1 and p2. This might not cause problems to the model as long as the data coherently transformed.
- In 'model.py', when changing temperature is involved. For function 'adjust_temp', 'pi_pdf -= pi_pdf.max()' may not be necessary as the final result is normalized by their sum [page 4]. For function 'sample_gaussian', s1 and s2 are multiplied by temp**2, which should have been temp**0.5 according to the paper [page 4]. Does this influence the results? Not sure?
- In 'main.py', in the main training loop, eta_step is unchanged while it should vary from eta_min to 1 [page 12], should replace R_step by R**epoch?

**Comment by mlelarge** [comment]
Thanks for your issues. Some answers below:
- I think the output of `skrnn_sample` is `(64,3)` and the last dimension is for encoding the pen (writing =0 or not =1, in this last case, we do a cut)
- the [`pi_pdf -= pi_pdf.max()`](https://github.com/dataflowr/Project-DL-Seq2Seq/blob/2a30984dfc2460f4dac98ce93b065aa6fb8100e8/sketch_generation/model.py#L167) is only here for numerical stability see [log-sum exp trick](https://en.wikipedia.org/wiki/LogSumExp#log-sum-exp_trick_for_log-domain_calculations)
- it looks like [`eta_step` is modified](https://github.com/dataflowr/Project-DL-Seq2Seq/blob/2a30984dfc2460f4dac98ce93b065aa6fb8100e8/sketch_generation/main.py#L99)
